### ISSUE TITLE
GDS Orb does not change when client connection drops

### ISSUE THREAD

**Issue by saba-ja** [nasa/fprime#473]
| | |
|:---|:---|
|**_F´ Version_**| 1.5|
|**_Affected Component_**| GDS  |
---
## Problem Description
Current implementation of GDS uses a `green orb` and a `red cross` symbol to show if there is any new communication between GDS and FSW. The orb changes to the cross when there is no update in the `channel` or `event` history in the last `config.dataTimeout` seconds. 

The issue is the orb does not change its state when client connection is dropped as shown below:

Although connection is dropped the orb is still being shown:

![image](https://user-images.githubusercontent.com/35859004/114260513-2e85f680-998a-11eb-8558-309e3b2f5205.png)

## How to Reproduce

1. Start GDS GUI without the client (`fprime-gds -n`) 
2. Start the Ref client (e.g. from the Ref directory on Mac run `./build-artifacts/Darwin/bin/Ref -a 0.0.0.0 -p 50000`)
3. Add a few `CMD_NO_OP` from `Commanding` page
3. Drop Ref connection.

## Expected Behavior
The orb should change to the cross symbol indicating the connection has dropped.

**Comment by saba-ja** [comment]
## Root cause

The orb state gets updated in the following lines:

https://github.com/nasa/fprime/blob/98f50f21cef5dfa1787e0c22dcf2193f7672c751/Gds/src/fprime_gds/flask/static/js/vue-support/tabetc.js#L59-L69

The `this.active` is an array with 2 boolean elements. The first element shows if there has been any new activity in channels table, and the second element indicates any new update in the `events` table.

The values of `active` array gets updated in the following function:

https://github.com/nasa/fprime/blob/98f50f21cef5dfa1787e0c22dcf2193f7672c751/Gds/src/fprime_gds/flask/static/js/datastore.js#L123-L133

The function gets called every [`config["dataPollIntervalMs"]`](https://github.com/nasa/fprime/blob/98f50f21cef5dfa1787e0c22dcf2193f7672c751/Gds/src/fprime_gds/flask/static/js/config.js
). The issue here is the `if statement` on line 126. The state of the `this.active` element will not get updated if there is no `new_items`.

## Correction

The following change in [fprime/Gds/src/fprime_gds/flask/static/js/datastore.js](https://github.com/nasa/fprime/blob/98f50f21cef5dfa1787e0c22dcf2193f7672c751/Gds/src/fprime_gds/flask/static/js/datastore.js) could resolve the issue:

```javascript
    updateActivity(new_items, index) {
        // Set active items, and register a timeout to turn it off again
        if (new_items.length > 0) {
            this.active.splice(index, 1, true);
            let timeout = config.dataTimeout * 1000;
            clearTimeout(this.active_timeout);
            this.active_timeout = setTimeout(() => this.active.splice(index, 1, false), timeout);
        } else {
            this.active.splice(index, 1, false);
        }
    }
```
We could argue the `setTimeout` is not needed since this function gets called back every `dataPollIntervalMs = 1000` milliseconds. If the `dataTimeout = 5` seconds the timeout gets cleared by `clearTimeout` before it gets to run.

**Comment by saba-ja** [comment]
Fixed in PR #475
